### PR TITLE
Support Venom codes

### PR DESCRIPTION
**Pull request by robertdown**
Allow [VeNom Codes](http://venomcoding.org/) to be imported as an external data set. Not really sure the best way to go about this, I had to test against 6.0.0 and would eventually like to be able to move this to a proper custom module.

### CODE REVIEW

**Comment by robertdown** [comment]
This code won't actually work - it needs 3 tables to be installed, which I've put in a custom module. For reference though, here's the SQL.

If the consensus is to bring the support into core, I'll move this SQL into database.sql, if we decide the custom module is the better option, I'll likely just withdraw the PR

```
--
-- Table structure for table `venom_codes`
--

DROP TABLE IF EXISTS `venom_dx`;
CREATE TABLE `venom_dx` (
  `dict_id` int,
  `term` varchar(255),
  `approved` int default NULL,
  `active` int default NULL,
  `subset_id` int default NULL,
  `subset` varchar(255) default NULL,
  `first_release` varchar(3) default NULL,
  `top_level_model` text default NULL,
  `large` tinyint default NULL,
  `small` tinyint default NULL,
  `farm` tinyint default NULL,
  `exotic` tinyint default NULL,
  `equine` tinyint default NULL,
  `revision` int default 0,
  PRIMARY KEY  (`dict_id`)
) ENGINE=InnoDB;

DROP TABLE IF EXISTS `venom_dx_test`;
CREATE TABLE `venom_dx_test` (
  `dict_id` int,
  `term` varchar(255),
  `approved` int default NULL,
  `active` int default NULL,
  `subset_id` int default NULL,
  `subset` varchar(255) default NULL,
  `first_release` varchar(3) default NULL,
  `top_level_model` text default NULL,
  `large` tinyint default NULL,
  `small` tinyint default NULL,
  `farm` tinyint default NULL,
  `exotic` tinyint default NULL,
  `equine` tinyint default NULL,
  `revision` int default 0,
  PRIMARY KEY  (`dict_id`)
) ENGINE=InnoDB;

DROP TABLE IF EXISTS `venom_proc`;
CREATE TABLE `venom_proc` (
  `dict_id` int,
  `term` varchar(255),
  `approved` int default NULL,
  `active` int default NULL,
  `subset_id` int default NULL,
  `subset` varchar(255) default NULL,
  `first_release` varchar(3) default NULL,
  `top_level_model` text default NULL,
  `large` tinyint default NULL,
  `small` tinyint default NULL,
  `farm` tinyint default NULL,
  `exotic` tinyint default NULL,
  `equine` tinyint default NULL,
  `revision` int default 0,
  PRIMARY KEY  (`dict_id`)
) ENGINE=InnoDB;
```

**Comment by stephenwaite** [comment]
:poodle:

File: library/standard_tables_capture.inc [hunk 1]
```diff
@@ -11,12 +11,16 @@
...
  * @author    Stephen Waite <stephen.waite@cmsvt.com>
+ * @author    Robert Down <robertdown@live.com>
  * @copyright Copyright (c) 2011 Phyaura, LLC <info@phyaura.com>
  * @copyright Copyright (c) 2019 Brady Miller <brady.g.miller@gmail.com>
  * @copyright Copyright (c) 2019 Stephen Waite <stephen.waite@cmsvt.com>
+ * @copyright Copyright (c) 2021 Robert Down <robertdown@live.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
+ use PhpOffice\PhpSpreadsheet\Reader\Xlsx;
```
Comment (bradymiller) [inline library/standard_tables_capture.inc:22]: wooohoooo, i think this is the first use of PhpOffice that will get into core

2DV

File: sql/database.sql [hunk 2]
```diff
@@ -8976,6 +8976,8 @@ INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel,
...
 INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('PHIN Questions', 111, 111, 0, '', 0, 0, 1, 0, 1, 'PHIN Questions', 0, 0, 0, 0, 0);
 INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('NCI-CONCEPT-ID', 112, 112, 0, '', 0, 0, 1, 0, 1, 'NCI CONCEPT ID', 0, 0, 0, 0, 0);
+INSERT INTO `code_types` (`ct_key`, `ct_id`, `ct_seq`, `ct_mod`, `ct_just`, `ct_mask`, `ct_fee`, `ct_rel`, `ct_nofs`, `ct_diag`, `ct_active`, `ct_label`, `ct_external`, `ct_claim`, `ct_proc`, `ct_term`, `ct_problem`, `ct_drug`) VALUES ('VALUESET', '113', '113', '0', '', '', '1', '1', '0', '1', '1', 'CQM Valueset', '13', '1', '1', '1', '1', '1');
+INSERT INTO code_types (ct_key, ct_id, ct_seq, ct_mod, ct_just, ct_fee, ct_rel, ct_nofs, ct_diag, ct_active, ct_label, ct_external, ct_claim, ct_proc, ct_term, ct_problem ) VALUES ('VENOM', 114, 114, 0, '', 1, 0, 0, 1, 1, 'VeNom Coding', 1, 1, 1, 1, 1);
```
Comment (bradymiller) [inline sql/database.sql:8980]: not seeing how this is being connected to the ct_external. also, i think there will need to be 3 entries.

**Comment by bradymiller** [comment]
those sql tables look fine to bring in (change the `top_level_model` in each of those tables to just `text` without the default stuff)

**Comment by bradymiller** [comment]
I just realized this is based on rel-600. rec working from master first and then bringing to rel-600 when it is in master.

**Comment by robertdown** [comment]
@bradymiller yeah, this wasn't approached very well. I had a lot of issues between master and 600, Still trying to untangle this.

**Comment by robertdown** [comment]
See #4800

**Pull request closed without merging**